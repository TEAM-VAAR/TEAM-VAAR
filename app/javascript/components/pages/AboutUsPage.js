import React, { Component, useState } from "react";
import {
  Carousel,
  CarouselItem,
  CarouselControl,
  CarouselIndicators,
  CarouselCaption,
} from "reactstrap";

class AboutUs extends Component {
  render() {
    return (
      <>
      <h2>About Us</h2>
        <AboutUsCarousel />
        <div>Footer</div>
      </>
    );
  }
}

export default AboutUs;

const items = [
  {
    src: "https://picsum.photos/id/700/1200/400",
    altText: "Project Manager",
    caption: "Alvin Accad",
    key: 1,
    linkedin: <a href="https://www.linkedin.com/in/alvin-a-a1b826189/">linkedin</a>,
    GitHub: <a href=" https://github.com/alvingithub1">GitHub</a>,
    description:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum",
  },
  {
    src: "https://picsum.photos/id/678/1200/400",
    altText: "Product Manager",
    caption: "Alexander Ballou",
    key: 2,
    linkedin: <a href="https://www.linkedin.com/in/alexander-ballou/">linkedin</a>,
    GitHub: <a href="https://github.com/necessities">GitHub</a>,
    description:
      "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?",
  },
  {
    src: "https://picsum.photos/id/456/1200/400",
    altText: "Code Lead",
    caption: "Ricky Fort",
    key: 3,
    linkedin: <a href="https://www.linkedin.com/in/ariseson/">linkedin</a>,
    GitHub: <a href="https://github.com/AriseSon">GitHub</a>,
    description:
      "At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.",
  },
  {
    src: "https://picsum.photos/id/500/1200/400",
    altText: "Design Lead",
    caption: "Venessa Palmer",
    key: 4,
    linkedin: <a href="https://www.linkedin.com/in/venessa-palmer-855038225/">linkedin</a>,
    GitHub: <a href="https://github.com/VenessaSP">GitHub</a>,
    description:
      "Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem",
  },
];

function AboutUsCarousel(args) {
  const [activeIndex, setActiveIndex] = useState(0);
  const [animating, setAnimating] = useState(false);

  const next = () => {
    if (animating) return;
    const nextIndex = activeIndex === items.length - 1 ? 0 : activeIndex + 1;
    setActiveIndex(nextIndex);
  };

  const previous = () => {
    if (animating) return;
    const nextIndex = activeIndex === 0 ? items.length - 1 : activeIndex - 1;
    setActiveIndex(nextIndex);
  };

  const goToIndex = (newIndex) => {
    if (animating) return;
    setActiveIndex(newIndex);
  };

  const slides = items.map((item) => {
    return (
      <CarouselItem
        onExiting={() => setAnimating(true)}
        onExited={() => setAnimating(false)}
        key={item.src}
      >
        <p className="description"> {item.description}</p>
        <h3 className="linkedin"> {item.linkedin}</h3>
        <p className="GitHub"> {item.GitHub}</p>
        

        <img src={item.src} alt={item.altText} />

        <CarouselCaption
          captionText={item.altText}
          captionHeader={item.caption}
        />
      </CarouselItem>
    );
  });

  return (
    <Carousel
      interval={false}
      activeIndex={activeIndex}
      next={next}
      previous={previous}
      {...args}
    >
      <CarouselIndicators
        items={items}
        activeIndex={activeIndex}
        onClickHandler={goToIndex}
      />
      {slides}
      <CarouselControl
        direction="prev"
        directionText="Previous"
        onClickHandler={previous}
      />
      <CarouselControl
        direction="next"
        directionText="Next"
        onClickHandler={next}
      />
    </Carousel>
  );
}
