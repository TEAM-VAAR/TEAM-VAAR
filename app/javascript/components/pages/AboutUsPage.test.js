import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import AboutUsPage from './AboutUsPage'
Enzyme.configure({ adapter: new Adapter() })

describe("When AboutUs renders", () => {
    it("displays a heading", () => {
      const aboutUsPage = shallow(<AboutUsPage />)
      const aboutUsPageHeading = aboutUsPage.find("h2")
      const aboutUsPageBody = aboutUsPage.find("items")
      const aboutUsPageCarousel = aboutUsPage.find("AboutUsCarousel")
  
      expect(aboutUsPageHeading.text()).toEqual("About Us")  
      expect(aboutUsPage.html()).toMatchSnapshot()
      
    })
  })