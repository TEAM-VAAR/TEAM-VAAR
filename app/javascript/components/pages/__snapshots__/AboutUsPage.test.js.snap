// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When AboutUs renders displays a heading 1`] = `"<h2>About Us</h2><div class=\\"carousel slide\\"><div class=\\"carousel-indicators\\"><button aria-label=\\"Alvin Accad\\" data-bs-target=\\"true\\" type=\\"button\\" class=\\"active\\"></button><button aria-label=\\"Alexander Ballou\\" data-bs-target=\\"true\\" type=\\"button\\" class=\\"\\"></button><button aria-label=\\"Ricky Fort\\" data-bs-target=\\"true\\" type=\\"button\\" class=\\"\\"></button><button aria-label=\\"Venessa Palmer\\" data-bs-target=\\"true\\" type=\\"button\\" class=\\"\\"></button></div><div class=\\"carousel-inner\\"><div class=\\"carousel-item active\\"><p class=\\"description\\"> I am a full-stack developer with experience building websites and web applications. I enjoy working with others on solving problems and communicating what can be done to solve them.</p><h3 class=\\"linkedin\\"> <a href=\\"https://www.linkedin.com/in/alvin-a-a1b826189/\\">linkedin</a></h3><p class=\\"GitHub\\"> <a href=\\" https://github.com/alvingithub1\\">GitHub</a></p><img src=\\"https://avatars.githubusercontent.com/u/74055731?v=4\\" alt=\\"Project Manager\\"/><div class=\\"carousel-caption d-none d-md-block\\"><h3>Alvin Accad</h3><p>Project Manager</p></div></div><div class=\\"carousel-item\\"><p class=\\"description\\"> I am a full stack web developer and software engineer with a background in IT and DevOps. I have experience working for a wide variety of organizations from large federal level ones to small consulting teams. I spend a lot of my free time tinkering with computers and learning about technology.</p><h3 class=\\"linkedin\\"> <a href=\\"https://www.linkedin.com/in/alexander-ballou/\\">linkedin</a></h3><p class=\\"GitHub\\"> <a href=\\"https://github.com/necessities\\">GitHub</a></p><img src=\\"https://avatars.githubusercontent.com/u/73265363?v=4\\" alt=\\"Product Manager\\"/><div class=\\"carousel-caption d-none d-md-block\\"><h3>Alexander Ballou</h3><p>Product Manager</p></div></div><div class=\\"carousel-item\\"><p class=\\"description\\"> My name is Ricky Fort, my position in Team VAAR is the Tech Lead. During the process of our Honest Review Application, I displayed solid understanding of the technical needs of the project and maintained the teams highly motivated attitude.</p><h3 class=\\"linkedin\\"> <a href=\\"https://www.linkedin.com/in/ariseson/\\">linkedin</a></h3><p class=\\"GitHub\\"> <a href=\\"https://github.com/AriseSon\\">GitHub</a></p><img src=\\"https://ca.slack-edge.com/T04B40L2C-U03K7G8TYU9-c8c0f535ad1b-512\\" alt=\\"Code Lead\\"/><div class=\\"carousel-caption d-none d-md-block\\"><h3>Ricky Fort</h3><p>Code Lead</p></div></div><div class=\\"carousel-item\\"><p class=\\"description\\"> Currently I am a student at a boot-camp style web development school in San Diego,  LEARN Academy. Here I am acquiring the necessary skills to become a junior web-developer. With over 480 hours of intense tutelage from experienced senior software developers, I am eager to apply my newly acquired skillset and be a contribution to the team! For this project I was responsible for styling, I wanted to create a visual representation of what the app is all about! I want users to focus on what the app is intended for while putting fun stylistic designs. </p><h3 class=\\"linkedin\\"> <a href=\\"https://www.linkedin.com/in/venessa-palmer-855038225/\\">linkedin</a></h3><p class=\\"GitHub\\"> <a href=\\"https://github.com/VenessaSP\\">GitHub</a></p><img src=\\"https://avatars.githubusercontent.com/u/107439475?s=400&amp;u=1f2cea854675cd8c7bf65d929f070581aa24ace0&amp;v=4\\" alt=\\"Design Lead\\"/><div class=\\"carousel-caption d-none d-md-block\\"><h3>Venessa Palmer</h3><p>Design Lead</p></div></div></div><a class=\\"carousel-control-prev\\" style=\\"cursor:pointer\\" role=\\"button\\" tabindex=\\"0\\"><span class=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\"></span><span class=\\"visually-hidden\\">Previous</span></a><a class=\\"carousel-control-next\\" style=\\"cursor:pointer\\" role=\\"button\\" tabindex=\\"0\\"><span class=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\"></span><span class=\\"visually-hidden\\">Next</span></a></div>"`;
